### ISSUE TITLE
Fatal error in SpecialOWLExport

### ISSUE THREAD

**Issue by Nikerabbit** [SemanticMediaWiki/SemanticMediaWiki#3173]
### Setup and configuration

- SMW version: 2.5.6
- MW version: 1.32.0-alpha
- PHP version: 7.1.17-1+ubuntu16.04.1+deb.sury.org+1 (fpm-fcgi)
- DB system (MySQL, Blazegraph, etc.) and version: 10.0.34-MariaDB-0ubuntu0.16.04.1

### Issue
Fatal error in PHP log:

2018-06-01 13:20:36 web1.translatewiki.net translatewiki_net-bw_: [8f5dd3644f907828fa90fe94] https://translatewiki.net/w/i.php?title=Special:ExportRDF/Thread:User_talk:Movses/%C3%83%C2%90%C3%82%C2%BE%C3%83%C2%91%C3%82%C2%88%C3%83%C2%90%C3%82%C2%B8%C3%83%C2%90%C3%82%C2%B1%C3%83%C2%90%C3%82%C2%BA%C3%83%C2%90%C3%82%C2%B8_%C3%83%C2%91%C3%82%C2%81_%C3%83%C2%90%C3%82%C2%BF%C3%83%C2%91%C3%82?%C3%83%C2%90%C3%82%C2%B5%C3%83%C2%91%C3%82%C2%84%C3%83%C2%90%C3%82%C2%B8%C3%83%C2%90%C3%82%C2%BA%C3%83%C2%91%C3%82%C2%81%C3%83%C2%90%C3%82%C2%B0%C3%83%C2%90%C3%82%C2%BC%C3%83%C2%90%C3%82%C2%B8_%C3%83%C2%90%C3%82%C2%BF%C3%83%C2%91%C3%82?%C3%83%C2%90%C3%82%C2%B8_%C3%83%C2%90%C3%82%C2%BF%C3%83%C2%90%C3%82%C2%B5%C3%83%C2%91%C3%82?%C3%83%C2%90%C3%82%C2%B5%C3%83%C2%90%C3%82%C2%B2%C3%83%C2%90%C3%82%C2%BE%C3%83%C2%90%C3%82%C2%B4%C3%83%C2%90%C3%82%C2%B0%C3%83%C2%91%C3%82%C2%85/reply&xmlmime=rdf   Error from line 184 of /srv/mediawiki/tags/2018-05-23_19:00:54/extensions/SemanticMediaWiki/includes/specials/SMW_SpecialOWLExport.php: Call to a member function isRedirect() on null

No stracktrace available.

### Steps to reproduce
When visiting the above URL. It's unclear what causes such url in the first place.

**Comment by kghbln** [comment]
This fix could probably be back-ported to 2.5.7

**Comment by mwjames** [comment]
The cause is a length issue that hits the 255 restriction and produces a NULL on a title longer than 255, as for comparison:

- `Special:ExportRDF/Foo/ÃÂ¾ÃÂÃÂ¸ÃÂ±ÃÂºÃÂ¸_ÃÂ_ÃÂ¿ÃÂ%3FÃÂµÃÂÃÂ¸ÃÂºÃÂÃÂ°ÃÂ¼ÃÂ¸_ÃÂ¿ÃÂ%3FÃÂ¸_ÃÂ¿ÃÂµÃÂ%3FÃÂµÃÂ²ÃÂ¾ÃÂ´ÃÂ°ÃÂ/reply&xmlmime=rdf` passes (253) while
- `Special:ExportRDF/Foo/ÃÂ¾ÃÂÃÂ¸ÃÂ±ÃÂºÃÂ¸_ÃÂ_ÃÂ¿ÃÂ%3FÃÂµÃÂÃÂ¸ÃÂºÃÂÃÂ°ÃÂ¼ÃÂ¸_ÃÂ¿ÃÂ%3FÃÂ¸_ÃÂ¿ÃÂµÃÂ%3FÃÂµÃÂ²ÃÂ¾ÃÂ´ÃÂ°ÃÂ---/reply&xmlmime=rdf` doesn't (256)

The proposed change should prevent the failure even though the length issue isn't addressed which itself cannot be solved by SMW.

**Comment by kghbln** [comment]
@mwjames I will do a pull tomorrow.

> The proposed change should prevent the failure even though the length issue isn't addressed which itself cannot be solved by SMW.

@Nikerabbit Worth creating a task for MW at Phabricator?

**Comment by mwjames** [comment]
> @Nikerabbit Worth creating a task for MW at Phabricator?

No, I think MW has a fixed length of 255 for a title and anything above will just blow.

**Comment by mwjames** [comment]
I'd say merge it and once on the sandbox test that the link (sandbox
link) from above no longer fails.

**Comment by kghbln** [comment]
> I'd say merge it and once on the sandbox test that the link (sandbox link) from above no longer fails.

Affirmative. We now get an RDF a expected.
